padinfo/padinfo.py

```diff
@@ -425,6 +442,23 @@ def findMonster(self, query, na_only=False):
     def _findMonster(self, query, na_only=False):
         monster_index = self.index_na if na_only else self.index_all
         return monster_index.find_monster(query)
+        
+    def findMonster2(self, query, na_only=False):
+        query = rmdiacritics(query)
+        nm, err, debug_info = self._findMonster2(query, na_only)
+
+        monster_no = nm.monster_no if nm else -1
+        self.historic_lookups[query] = monster_no
+        dataIO.save_json(self.historic_lookups_file_path, self.historic_lookups)

```

Comment:
You probably need a separate historic lookups file for id2 since it will return different values than id.

Reply:
saved as historic_lookups_file_path_id2 / historic_lookups_id2